test/jb_controller/burn_tokens_of.test.js

```diff
@@ -0,0 +1,343 @@
+  const MEMO = 'Test Memo';
+  const TOTAL_SUPPLY = 100000;
+  const AMOUNT_TO_BURN = 20000;
+  const RESERVED_RATE = 5000;
+  let BURN_INDEX;
+
+  before(async function () {
+    let jbOperationsFactory = await ethers.getContractFactory('JBOperations');
+    let jbOperations = await jbOperationsFactory.deploy();
+
+    BURN_INDEX = await jbOperations.BURN();
+  });
+
+  async function setup() {
+    let [deployer, projectOwner, holder, ...addrs] = await ethers.getSigners();
+
+    const blockNum = await ethers.provider.getBlockNumber();
+    const block = await ethers.provider.getBlock(blockNum);
+    const timestamp = block.timestamp;
+
+    let promises = [];
+
+    promises.push(deployMockContract(deployer, jbOperatoreStore.abi));
+    promises.push(deployMockContract(deployer, jbProjects.abi));
+    promises.push(deployMockContract(deployer, jbDirectory.abi));
+    promises.push(deployMockContract(deployer, jbFundingCycleStore.abi));
+    promises.push(deployMockContract(deployer, jbTokenStore.abi));
+    promises.push(deployMockContract(deployer, jbSplitsStore.abi));
+    promises.push(deployMockContract(deployer, jbToken.abi));
+
+    let [
+      mockJbOperatorStore,
+      mockJbProjects,
+      mockJbDirectory,
+      mockJbFundingCycleStore,
+      mockTokenStore,
+      mockSplitsStore,
+      mockToken,
+    ] = await Promise.all(promises);
+
+    let jbControllerFactory = await ethers.getContractFactory('JBController');
+    let jbController = await jbControllerFactory.deploy(
+      mockJbOperatorStore.address,
+      mockJbProjects.address,
+      mockJbDirectory.address,
+      mockJbFundingCycleStore.address,
+      mockTokenStore.address,
+      mockSplitsStore.address,
+    );
+
+    promises = [];
+
+    promises.push(mockJbProjects.mock.ownerOf.withArgs(PROJECT_ID).returns(projectOwner.address));
+
+    promises.push(
+      mockJbDirectory.mock.isTerminalDelegateOf.withArgs(PROJECT_ID, holder.address).returns(false),
+    );
+
+    promises.push(
+      mockJbDirectory.mock.isTerminalDelegateOf
+        .withArgs(PROJECT_ID, projectOwner.address)
+        .returns(false),
+    );
+
+    promises.push(
+      mockJbFundingCycleStore.mock.currentOf.withArgs(PROJECT_ID).returns({
+        // mock JBFundingCycle obj
+        number: 1,
+        configuration: timestamp,
+        basedOn: timestamp,
+        start: timestamp,
+        duration: 0,
+        weight: 0,
+        discountRate: 0,
+        ballot: ethers.constants.AddressZero,
+        metadata: packFundingCycleMetadata({
+          pauseBurn: 0,
+          pauseMint: 0,
+          reservedRate: RESERVED_RATE,
+        }),
+      }),
+    );
+
+    // only non-reserved are minted, minting total supply in holder account
+    promises.push(
+      mockTokenStore.mock.mintFor
+        .withArgs(
+          holder.address,
+          PROJECT_ID,
+          (TOTAL_SUPPLY * (10000 - RESERVED_RATE)) / 10000,
+          /*_preferClaimedTokens=*/ true,
```

Comment:
If this is true in every test, you may be better off using a `const PREFER_CLAIMED_TOKENS = true` so that you don't have to use the arg comment